### PR TITLE
meta: Add a GetDBMeta function to meta.

### LINKED ISSUE #53685: Add a function named GetDBMeta to meta.go.
## Enhancement

Add a function named `GetDBMeta` to `meta.go`. This function will allow the caller get all tableInfos of a database.

ref: https://github.com/pingcap/tiflow/issues/11109#:~:text=func%20(m%20*Meta)%20ListTablesByFn(dbID%20int64%2C%20fn%20func(info%20*model.TableNameInfo)%20bool)%20(%5B%5D*model.TableInfo%2C%20error)%20%7B

### PR DESCRIPTION
**Pull request by asddongmen**
### What problem does this PR solve?

Issue Number: close #53685 

Problem Summary:

### What changed and how does it work?

Add a function named `GetDBMeta` to meta.go. This function will allow the caller to get all raw tableInfos of database.

### Check List

Tests 

- [ ] Unit test
- [ ] Integration test
- [ ] Manual test (add detailed scripts or steps below)
- [x] No need to test
  > - [ ] I checked and no code files have been changed.
  > 
	Covered by the existing unit test cases. 

Side effects

Documentation

### Release note

Please refer to [Release Notes Language Style Guide](https://pingcap.github.io/tidb-dev-guide/contribute-to-tidb/release-notes-style-guide.html) to write a quality release note.

```release-note
None
```

### CODE REVIEW

File: pkg/meta/meta.go [hunk 1]
```diff
@@ -1022,6 +1022,49 @@ func (m *Meta) ListTables(dbID int64) ([]*model.TableInfo, error) {
 	return tables, nil
 }
 
+// ListTablesByFilter return all tables that does not filtered by filter in a database.
```
Comment (3AceShowHand) [inline pkg/meta/meta.go:1025]: please add a comment to note that this method is used by the ticdc, to prevent deleted

File: pkg/meta/meta.go [hunk 2]
```diff
@@ -1022,6 +1022,51 @@ func (m *Meta) ListTables(dbID int64) ([]*model.TableInfo, error) {
...
+		tbInfo.DBID = dbID
+
+		tables = append(tables, tbInfo)
+	}
+
+	return tables, nil
+}
+
 // ListSimpleTables shows all simple tables in database.
 func (m *Meta) ListSimpleTables(dbID int64) ([]*model.TableNameInfo, error) {
```
Comment (okJiang) [inline pkg/meta/meta.go:1071]: It is better to call ListTablesByFilter(dbID, nil) to implement ListSimpleTables

File: pkg/meta/meta.go [hunk 3]
```diff
@@ -1022,6 +1022,51 @@ func (m *Meta) ListTables(dbID int64) ([]*model.TableInfo, error) {
 	return tables, nil
 }
 
+// ListTablesByFilter return all tables that does not filtered by filter in a database.
+// Note(dongmen): This method is used by TiCDC to reduce the time of changefeed initialization.
+// Ref: https://github.com/pingcap/tiflow/issues/11109
+func (m *Meta) ListTablesByFilter(
```
Comment (okJiang) [inline pkg/meta/meta.go:1028]: it is better to use `ListTablesByFilter(dbID, nil)` to implement `ListTables`

File: pkg/meta/meta.go [hunk 4]
```diff
@@ -1022,6 +1022,51 @@ func (m *Meta) ListTables(dbID int64) ([]*model.TableInfo, error) {
...
+			continue
+		}
+		tbName := &model.TableNameInfo{}
+		err = json.Unmarshal(r.Value, tbName)
+		if err != nil {
+			return nil, errors.Trace(err)
+		}
+		if filter(tbName) {
+			continue
+		}
```
Comment (okJiang) [inline pkg/meta/meta.go:1055]: if filter != nil {
    ...
}

**Comment by D3Hunter** [comment]
what kind of filter will CDC use?

**Review by CharlesCheung96 — APPROVED** [review]

File: pkg/meta/meta.go [hunk 5]
```diff
@@ -989,17 +989,27 @@ func (m *Meta) IterTables(dbID int64, fn func(info *model.TableInfo) error) erro
 	return errors.Trace(err)
 }
 
-// ListTables shows all tables in database.
-func (m *Meta) ListTables(dbID int64) ([]*model.TableInfo, error) {
+// GetDBMeta return all meta information of a database.
+// Note(dongmen): This method is used by TiCDC to reduce the time of changefeed initialization.
+// Ref: https://github.com/pingcap/tiflow/issues/11109
+func (m *Meta) GetDBMeta(dbID int64) ([]structure.HashPair, error) {
```
Comment (D3Hunter) [inline pkg/meta/meta.go:995]:
```suggestion
func (m *Meta) GetMetasByDBID(dbID int64) ([]structure.HashPair, error) {
```
i don't have a better name, but i prev one doesn't desc what it does

File: pkg/meta/meta_test.go [hunk 6]
```diff
@@ -293,6 +293,7 @@ func TestMeta(t *testing.T) {
 		})
 		require.NoError(t, err)
 	}
+
```
Comment (D3Hunter) [inline pkg/meta/meta_test.go:296]:
```suggestion
```

**Review by D3Hunter — COMMENTED** [review]
rest lgtm

**Review by D3Hunter — APPROVED** [review]

**Review by wjhuang2016 — APPROVED** [review]

**Pull request merged**